File: average_height.py

odd = []
even = []
res = []
test_cases = int(input())
for j in range(test_cases):
    n = int(input())
    array = list(map(int, input().split()))

    for i in array:
        if i % 2 == 0:
            even.append(i)
        else:
            odd.append(i)

    print(*odd+even)
    odd.clear()
    even.clear()

File: average_height.cpp

# include <iostream>
# include <vector>

using namespace std;

int main()
{
    int test_cases, n, input;
    vector <int> odd, even;
    cin >> test_cases;
    int k = 1;
    while (k <= test_cases){
    cin >> n;

    for (int i = 0; i < n; i++){
        cin >> input;
        if (input % 2 == 0){
            even.push_back(input);
        }
        else if (input % 2 != 0){
            odd.push_back(input);
        }
        
    } 

    for (int i = 0; i < odd.size(); i++){
        cout << odd[i] << " ";
    }
    
    for (int i = 0; i < even.size(); i++){
        cout << even[i] << " ";
        
    }
    cout << endl;
    odd.clear();
    even.clear();
    k += 1;
    }
    return 0;
}
